torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -808,31 +808,34 @@ ProcessGroupNCCL::ProcessGroupNCCL(
   std::string torch_distributed_debug =
       getCvarString({"TORCH_DISTRIBUTED_DEBUG"}, OFF.c_str());
   LOG(INFO) << logPrefix() << "ProcessGroupNCCL initialization options: "
-            << "NCCL version: " << getNcclVersion() << ", size: " << size
-            << ", global rank: " << globalRank()
-            << ", TORCH_NCCL_ASYNC_ERROR_HANDLING: " << asyncErrorHandling_
-            << ", TORCH_NCCL_DUMP_ON_TIMEOUT: " << dumpOnException_
-            << ", TORCH_NCCL_WAIT_TIMEOUT_DUMP_MILSEC: "
-            << waitTimeoutDumpInMilSec_
-            << ", TORCH_NCCL_DESYNC_DEBUG: " << desyncDebug_
-            << ", TORCH_NCCL_ENABLE_TIMING: " << enableTiming_.load()
-            << ", TORCH_NCCL_BLOCKING_WAIT: " << blockingWait_
+            << "size: " << size << ", global rank: " << globalRank()
             << ", TIMEOUT(ms): " << options_->timeout.count()
             << ", USE_HIGH_PRIORITY_STREAM: "
             << options_->is_high_priority_stream
             << ", SPLIT_FROM: " << options_->split_from
             << ", SPLIT_COLOR: " << options_->split_color
-            << ", TORCH_DISTRIBUTED_DEBUG: " << torch_distributed_debug
+            << ", PG Name: " << options_->group_name;
+
+  if (rank_ % localDeviceCount_ == 0) {
+    LOG(INFO)
```

Comment:
skipping the nccl init logs seems particularly bad to me.  It could be useful to see a difference in env/config on different ranks.  (or, if different ranks create different PGs).

Reply:
If the user did not sandbox each process -- the env will be the same on that node.

If the user sandbox each process -- then most likely, CUDA_VISIBLE_DEVICES points to one local GPU, `rank_ % localDeviceCount_ == 0` will be always true and hence each process will print env in their own sandbox.

Reply:
That said, I can revert this change if it helps the warning dedup to land.

Reply:
But note that the change here purposely separate out env info and config info (i.e. options passed to PG)

Reply:
Unmuted env printout

Reply:
i can see this one both ways.

on one hand, we have the TORCH_LOGS_RANK env now, and savvy users would already be filtering this way to look at just their favorite rank.  On the cluster, this may be more of an issue, but it depends on how the UI does rank filtering there.

a related issue is that often there are many ProcessGroups created per process/global-rank.  If a model uses 50 or 100 PGs (i've seen this), then that's still 50 or  100 per rank, and we may need an additional solution (like ENV cache) for solving this.